python/tank/templatekey.py

```diff
@@ -256,6 +273,145 @@ def _as_string(self, value):
         return value if isinstance(value, basestring) else str(value)
 
 
+class TimestampKey(TemplateKey):
+    """
+    Key whose value is a time string formatted with strftime.
+    """
+
+    def __init__(
+        self,
+        name,
+        utc_default=False,
+        format_spec="%Y-%m-%d-%H-%M-%S"
+    ):
+        """
+        Constructor
+        :param name: Name by which the key will be refered.
+        :param utc_default: If True, the default value will be an UTC time. Defaults to False.
```

Comment:
Actually, having two classes Timestamp and UtcTimestamp doesn't gell well with the idea of having a non-now default value for a timestamp. See my comment at the comment on how the interface could be implemented.